src/Chainweb/Pact/PactService/ExecBlock.hs

```diff
@@ -424,7 +434,9 @@ applyPactCmd isGenesis env miner cmd = StateT $ \(T2 mcache maybeBlockGasRemaini
       else do
         pd <- getTxContext (publicMetaOf gasLimitedCmd)
         spv <- use psSpvSupport
-        liftIO $! applyCmd v logger gasLogger env miner (gasModel pd) pd spv gasLimitedCmd initialGas mcache
+        let timeoutError = TxTimeout (requestKeyToTransactionHash $ P.cmdToRequestKey cmd)
+        let txTimeout = maybe id (\limit -> maybe (throwM timeoutError) return <=< timeout (fromIntegral limit)) txTimeLimit
```

Comment:
this is extremely compressed code, break into cases or something

Reply:
Let me know what you think of the new version